spec/shared_stripe_examples/subscription_examples.rb

```diff
@@ -633,6 +633,17 @@
         expect(subscription.default_tax_rates.first.id).to eq(tax_rate.id)
       end
     end
+
+    it 'expands latest_invoice.payment_intent' do
```

Comment:
Once the modifications based on the subscription status are completed, I think you can add some test cases for each status.